app/src/composables/use-relation-multiple.ts

```diff
@@ -102,17 +102,18 @@ export function useRelationMultiple(
 			);
 			const deleteIndex = _value.value.delete.findIndex((id) => id === item[targetPKField]);
 
-			const updatedItem: Record<string, any> = {};
+			const updatedItem: Record<string, any> = cloneDeep(item);
 
 			if (editsIndex !== -1) {
-				merge(
+				mergeWith(
 					updatedItem,
-					cloneDeep(item),
 					{ $type: 'updated', $index: editsIndex, $edits: editsIndex },
-					_value.value.update[editsIndex]
+					_value.value.update[editsIndex],
+					(_objValue, srcValue) => {
+						// skip the default merge behavior for objects/arrays
+						if (isObjectLike(srcValue)) return srcValue;
```

Comment:
```suggestion
						if (Array.isArray(srcValue)) return srcValue;
```
If we don't merge objects the "lodash" way, we run into the issue that when you edit a field in a m2m and then revert the changes, the display value will disappear as it wont be merged.

https://user-images.githubusercontent.com/22577866/187885683-044ce707-45e0-424c-b977-4a60451c4685.mp4

Reply:
Ah... that is very true. This does feels like a catch-22 now, as the reasoning to not merge objects the "lodash" way was to fix cases such as this where it'll visually appear the "lodash" way, but the raw value isn't:

https://user-images.githubusercontent.com/42867097/187914816-82ceef6b-e29f-45ba-92b1-cb4b7383cf01.mp4

However I do agree that this is probably the edge case between the two, so perhaps we can move forward with only skipping arrays for now and try to find a fix for this elsewhere?

Reply:
Hmm, didn't think of that one. I guess the right fix would be to lodash merge the top level and the field with nested object values inside it and the rest should be just replaced. 🤔

Reply:
Something like this could maybe fix both situations.
```js
updatedItem = {...cloneDeep(item), $type: 'updated', $index: editsIndex, $edits: editsIndex, ..._value.value.update[editsIndex]}

if(isM2M || isM2A) {
  updatedItem = {...updatedItem, [junctionField]: {...updatedItem, ..._value.value.update[editsIndex][junctionField] }}
}
```

Reply:
I tried giving a stab at that, but it seems like arrays are still not playing nicely 🤔 but it could be an issue with my implementation. Here's what I attempted based on your suggestion:

```js
			let updatedItem: Record<string, any> = cloneDeep(item);

			if (editsIndex !== -1) {
				updatedItem = {
					...updatedItem,
					$type: 'updated',
					$index: editsIndex,
					$edits: editsIndex,
					..._value.value.update[editsIndex],
				};

				if (relation.value?.type === 'm2m' || relation.value?.type === 'm2a') {
					updatedItem = {
						...updatedItem,
						[relation.value.junctionField.field]: {
							..._value.value.update[editsIndex][relation.value.junctionField.field],
						},
					};
				}
			}
```

Reply:
I played a bit around with it and I think I have both problems solved though I don't know why the code you wrote doesn't work and mine which is really similar does. Feel free to play around with my solution as im not 100% confident at the moment that I fixed everything and will test that tomorrow a bit more.

Reply:
Thanks! Interestingly I also can't see a big difference aside from the update within isM2M and isM2A, but your current solution definitely works to resolve all the cases now ❤

Tested M2M display (as you pointed out in this comment thread), json editing and repeater deletions (which was pointed out in the comment thread here: https://github.com/directus/directus/pull/13574#discussion_r882708057):

https://user-images.githubusercontent.com/42867097/188048726-f0a8e496-5dd2-475c-8e24-65894a71e419.mp4